resilience4j-circuitbreaker/src/main/java/io/github/resilience4j/circuitbreaker/internal/ForcedOpenState.java

```diff
@@ -0,0 +1,74 @@
+/*
+ *
+ *  Copyright 2016 Robert Winkler
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *
+ */
+package io.github.resilience4j.circuitbreaker.internal;
+
+import io.github.resilience4j.circuitbreaker.CircuitBreaker;
+import io.github.resilience4j.circuitbreaker.CircuitBreakerConfig;
+
+final class ForcedOpenState extends CircuitBreakerState {
+
+    private final CircuitBreakerMetrics circuitBreakerMetrics;
+
+    ForcedOpenState(CircuitBreakerStateMachine stateMachine) {
+        super(stateMachine);
+        CircuitBreakerConfig circuitBreakerConfig = stateMachine.getCircuitBreakerConfig();
+        final int size = circuitBreakerConfig.getRingBufferSizeInHalfOpenState();
+        this.circuitBreakerMetrics = new CircuitBreakerMetrics(size);
+    }
+
+    /**
+     * Returns false, if the wait duration has not elapsed.
+     * Returns true, if the wait duration has elapsed and transitions the state machine to HALF_OPEN state.
+     *
+     * @return false, if the wait duration has not elapsed. true, if the wait duration has elapsed.
+     */
+    @Override
+    boolean isCallPermitted() {
+        return false;
```

Comment:
Do you think we should measure metrics here? It could be interesting how many calls have been rejected.
See OpenState
`circuitBreakerMetrics.onCallNotPermitted();`

Reply:
Do we? What is the usual behaviour in other CircuitBreakers? But then why not measure how many we allowed when we are `DISABLED`? Should then have a difference between `DISABLED` and `FORCED_CLOSED`?

Reply:
Shouldn't be better if `FORCED_CLOSED` was a real state, like `FORCED_OPEN` where both record metrics but do not generate events, while being disabled/enabled simply switches off/on the CB but does not change state or update metrics? So the cb is disabled -> it is like it is bypassed and no call arrives to it, but keeps the same state and metrics as before and resumes from that point when it is enabled?

Reply:
I think `DISABLED` should be a state where everything (metrics and events) is disabled. When I disable a CB I don't care about metrics anymore. It would be even confusing when metrics still change in my monitoring system. Since there is no `ON` state, the user can choose to transition either to `OPEN`, `FORCED_OPEN`, `HALF_OPEN` or  `CLOSED`. Therefore I don't think that it makes sense to keep the metrics of the previous state.
The state `FORCED_OPEN` is somehow similar to `OPEN` but does not change back to `HALF_OPEN`. But I would expect to see metrics (number of rejected calls) in my monitoring system.

Reply:
K, I will implement it in the weekend